test_scripts/API/VehicleData/WindowStatus/common.lua

```diff
@@ -0,0 +1,454 @@
+    { location = { CoL = 49, row = 49 },
+      state = { approximatePosition = 50, deviation = 50 }
+    }
+  },
+  ["invalidName_row"] = { -- invalid name for row parameter from Grid structure
+    { location = { col = 49, RoW = 49 },
+      state = { approximatePosition = 50, deviation = 50 }
+    }
+  },
+  ["empty_state"] = { -- empty state parameter
+    { location = { col = 49, row = 49 },
+      state = {}
+    }
+  },
+  ["invalidType_state"] = { -- invalid type for state parameter
+    { location = { col = 49, row = 49 },
+      state = "string"
+    }
+  },
+  ["missing_state"] = { -- without state parameter
+    { location = { col = 49, row = 49 } }
+  },
+  ["invalidName_state"] = { -- invalid name for state parameter
+    { location = { col = 49, row = 49 },
+      StaTe = { approximatePosition = 50, deviation = 50 }
+    }
+  },
+  ["invalidName_approximatePosition"] = { -- invalid name for approximatePosition parameter from WindowState structure
+    { location = { col = 49, row = 49 },
+      state = { ApproximatePositioN = 50, deviation = 50 }
+    }
+  },
+  ["invalidName_deviation"] = { -- invalid name for deviation parameter from WindowState structure
+    { location = { col = 49, row = 49 },
+      state = { approximatePosition = 50, DeviatioN = 50 }
+    }
+  }
+}
+
+--[[ Functions ]]
+
+--[[ @updatePreloadedPT: Update preloaded file with additional permissions for WindowStatus
+--! @parameters: none
+--! @return: none
+--]]
+function m.updatePreloadedPT()
+  m.backupPreloadedPT()
+  local pt = m.getPreloadedPT()
+  local WindowStatusGroup = {
+    rpcs = {
+      GetVehicleData = {
+        hmi_levels = {"BACKGROUND", "FULL", "LIMITED"},
+        parameters = {"windowStatus"}
+      },
+      SubscribeVehicleData = {
+        hmi_levels = {"BACKGROUND", "FULL", "LIMITED"},
+        parameters = {"windowStatus"}
+      },
+      OnVehicleData = {
+        hmi_levels = {"BACKGROUND", "FULL", "LIMITED"},
+        parameters = {"windowStatus"}
+      },
+      UnsubscribeVehicleData = {
+        hmi_levels = {"BACKGROUND", "FULL", "LIMITED"},
+        parameters = {"windowStatus"}
+      }
+    }
+  }
+  pt.policy_table.app_policies["default"].groups = { "Base-4", "WindowStatus" }
+  pt.policy_table.functional_groupings["WindowStatus"] = WindowStatusGroup
+  pt.policy_table.functional_groupings["DataConsent-2"].rpcs = json.null
+  m.setPreloadedPT(pt)
+end
+
+--[[ @preconditions: Clean environment and backup sdl_preloaded_pt.json file
+--! @parameters:
+--! isPreloadedUpdate: if true then sdl_preloaded_pt.json file will be updated, otherwise - false
+--! @return: none
+--]]
+function m.preconditions(isPreloadedUpdate)
+  if isPreloadedUpdate == nil then isPreloadedUpdate = true end
+  actions.preconditions()
+  if isPreloadedUpdate == true then
+    m.backupPreloadedPT()
+    m.updatePreloadedPT()
+  end
+end
+
+--! @pTUpdateFunc: Policy Table Update with allowed "Base-4" group for application
```

Comment:
`Base-4` is assigned to the app by default. Most interesting that new group is assigned to the app.